### PR TITLE
Refactor TransferResponses into LedgerMovements

### LINKED ISSUE #5925: Tidy up LedgerEntry messages in TransferResponse events
Currently there is a `reference` field and a `type` field which are both strings and and in a random event I got from a system test run look  backwards to me. 

image

Looking at a few in the DB seems to show inconsistent usage:

image

In particular this lower case 'settlement' seems a bit odd. It would be good to clarify the purpose of these fields and make their usage consistent.

`type` should probably be an ENUM rather than a string.

Bonus Task:

In the datanode we currently have to decipher which accounts `(party, asset, market, type)` a ledger entry is referring to by decoding the `accountId` which is a string concatenation with special characters for no market/system owner and this is rather fragile and relies on quirks like party IDs being 64 characters long. It would be better to have explicit  `(optional) party_id`, `(optional) market_id`, `(required) asset_id`, `(required) account_type` fields

### PR DESCRIPTION
**Pull request by jeremyletang**
Rename TransferResponse into LedgerMovement.
Use AccountDetails type instead of account IDs or full account struct. 

Barely touched the datanode part, just enough so it can work and do not annoy too much with other PR in progress.

close #5925

### CODE REVIEW

**Review by pscott31 — DISMISSED** [review]
Looks very sensible to me!

**Comment by jeremyletang** [comment]
```
{
"system-tests": "feature/transfer-response-to-ledger-movements"
}
```

**Review by witgaw — DISMISSED** [review]
Verified test change looks legit.

**Comment by AnExsomnis** [comment]
IMO good to go!

**Review by AnExsomnis — DISMISSED** [review]

**Review by witgaw — DISMISSED** [review]

File: core/types/collateral.go [hunk 1]
```diff
@@ -28,6 +80,15 @@ type Account struct {
 	Type     AccountType
 }
 
+func (a Account) ToDetails() *AD {
+	return &AD{
```
Comment (ValentinTrinque) [inline core/types/collateral.go:84]: Can we name AD as AccountDetails ?

Comment (jeremyletang) [reply core/types/collateral.go:84]: We can. I just liked better AD as it’s shorted and I thought I’d use it more in the codebase / it looks slightly like ID as well, which was it relates too

Comment (jeremyletang) [reply core/types/collateral.go]: addressed.

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**